app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php

```diff
@@ -2021,11 +2021,7 @@ protected function _applyProductLimitations()
             $this->getConnection()->quoteInto('cat_index.store_id=?', $filters['store_id'], 'int'),
         ];
         if (isset($filters['visibility']) && !isset($filters['store_table'])) {
-            $conditions[] = $this->getConnection()->quoteInto(
```

Comment:
Still not reverted here, please use the mor ereadable version with multiple lines